scipy/optimize/tests/test_zeros.py

```diff
@@ -963,3 +978,23 @@ def test_maxiter_int_check_gh10236(method):
     message = "'float' object cannot be interpreted as an integer"
     with pytest.raises(TypeError, match=message):
         method(f1, 0.0, 1.0, maxiter=72.45)
+
+def test_bisect_special_parameter():
+    # give some zeros method strange parameters
+    # and check whether an exception appears
+    root = 0.1
+    args = (1e-09, 0.004, 10, 0.27456)
+    rtolbad = 4 * np.finfo(float).eps / 2
+
+    def f(x):
+        return x - root
+
+    methods = [zeros.bisect, zeros.ridder, zeros.brentq, zeros.brenth]
+
+    for method in methods:
+        with pytest.raises(ValueError) as excinfo:
+           _ = method(f, -1e8, 1e7, args=args, xtol=-1e-6, rtol=TOL)
+        assert "xtol too small" in excinfo.exconly()
+        with pytest.raises(ValueError) as excinfo:
+           _ = method(f, -1e8, 1e7, args=args, xtol=1e-6, rtol=rtolbad)
+        assert "rtol too small" in excinfo.exconly()
```

Comment:
We generally prefer `pytest.mark.pytest.mark.parametrize` rather than a for loop
```suggestion
@pytest.mark.pytest.mark.parametrize("method", [zeros.bisect, zeros.ridder, zeros.brentq, zeros.brenth])
def test_bisect_special_parameter(method):
    # give some zeros method strange parameters
    # and check whether an exception appears
    root = 0.1
    args = (1e-09, 0.004, 10, 0.27456)
    rtolbad = 4 * np.finfo(float).eps / 2

    def f(x):
        return x - root

    with pytest.raises(ValueError, match="xtol too small"):
       method(f, -1e8, 1e7, args=args, xtol=-1e-6, rtol=TOL)
    with pytest.raises(ValueError, match="rtol too small"):
       method(f, -1e8, 1e7, args=args, xtol=1e-6, rtol=rtolbad)
```